### ISSUE TITLE
Right-to-left functionality missing in notification bubbles.

### ISSUE THREAD

**Issue by Hew-ux** [batocera-linux/batocera-emulationstation#1140]
![image](https://user-images.githubusercontent.com/67527064/154590182-09b77f1a-158e-4cac-b2bc-60ecaca85065.png)
For reference, the notification bubble at the top _should_ be reading:
v34-dev 2022/02/16 10:52 .ﺮﻓﻮﺘﻣ ﺚﻳﺪﺤﺗ ﺪﺟﻮﻳ
but it is doing so in reverse order. Each individual character has been printed in left-to-right, when it should have been printed in right-to-left.

Slightly related issue: cursive language seems to be impossible in ES's menus. This would make reading it easier if supported.
![image](https://user-images.githubusercontent.com/67527064/154590558-83932579-5445-45b0-b21a-3f59613666b6.png)
I experimented with using different fonts that are specifically made for Arabic, but it seems to always want to insert spaces no matter what.

**Comment by Chipsum** [comment]
Unless some Dev does something about it, We might find ourselves in the only choice, where we convert all characters to Arabic Presentation Forms-B, and reverse every phrase.

**Comment by Hew-ux** [comment]
The problem is specifically about notification bubbles though, so you wouldn't be reversing every phrase. You would only be reversing the phrases which appear in those notification bubbles.

The missing cursive type is also a problem which applies to all languages, but it dearly affects the readability of Arabic more so than others.

**Comment by Chipsum** [comment]
Yes but it's solvable through Forms-B, this is Mesen's Emulator OSD messages with normal Arabic Unicode Range 600

![image](https://user-images.githubusercontent.com/37517008/221797711-20030ee3-5b0e-4bb1-ba91-db6ffad90da9.png)


Meanwhile, this is with Forms-B range + reversed
![image](https://user-images.githubusercontent.com/37517008/221797816-f5eabe3e-6dca-40db-a01f-5e8d0ff1b05d.png)

**Comment by Chipsum** [comment]
This is how it works on emulationstation, IF the font did contain all Forms-B characters:
this is for the word QUIT, خروج
![image](https://user-images.githubusercontent.com/37517008/221802920-5ccebaef-c293-4720-832f-f25972894105.png)
We can see here that the first two letters of the word are correctly displayed, but the other two they are not present in the font (which I will try and learn how to change it and see the results).
you can use this to convert a normal set of Arabic into Forms-B reversed and try yourself:
https://omar84.com/docs/taw/arabic_writer.html

**Comment by Chipsum** [comment]
Yeeeeeeeap, EmulationStation uses the font Cairo which I really did distinguish, it's a famous Arabic font (not one of the best at least for me).

it lacks the remaining Forms-B Characters:
![image](https://user-images.githubusercontent.com/37517008/221806819-8daf8e07-a0d2-45f8-b122-47a198610ed2.png)

I will try to replace it with another font that contains all Forms-B characters and see.

**Comment by Chipsum** [comment]
and that's the result :)

![image](https://user-images.githubusercontent.com/37517008/221808388-5829cbdd-f663-4774-9779-72d40fbaab3f.png)

**Comment by Hew-ux** [comment]
I see, sorry for misunderstanding what you said and thank you for researching this. It may be worth making into its own issue report at this point, as this one was mainly about the RTL function not working correctly, the cursive writing was something noted by another user which I added mention of.

It should be harmless to replace the current font with Forms-B, if you'd like to do that in the meantime. After all, this issue report has remained stagnant for over a year now, so I don't think this is high-priority for the devs to fix.

**Comment by Chipsum** [comment]
@Hew-ux Yeah, I strongly recommend the RTL thing, Hope devs will work on both the cursive and the RTL.

**Comment by Abdullah-Albanna** [comment]
@Chipsum 
Could you provide more information about how you fixed it, I want to fix mine too.

In which file did you edit
What is the tools that you use to fix it and how

**Comment by DubaiShark** [comment]
> and that's the result :)
> 
> ![image](https://user-images.githubusercontent.com/37517008/221808388-5829cbdd-f663-4774-9779-72d40fbaab3f.png)
> 
> But that's a Plan B solution, solving the issue without using Forms-B reversed is better and it would even facilitate the translation.

Hi

Can you tell us what was the Plan B solution so we can correct the spaces between the letters


Thanks

**Comment by nadenislamarre** [comment]
LC_MESSAGES/emulationstation2.po:msgid "UPDATING %s"
LC_MESSAGES/emulationstation2.po-msgstr "تحديث %s"

in fact, this is how the translator set it.
this is why it is reversed.
just fix the translation on http://translations.batocera.org/ and it will be ok.
i think we can close.
